packages/qwik-city/runtime/src/url-params-encoding.ts

```diff
@@ -0,0 +1,262 @@
+/**
+ * Encode a params into a URL query string.
+ *
+ * The encoding is done to remain human readable, but at the same time can deal with complex types
+ * such as `Date`, `Object` and `Array`.
+ *
+ * @public
+ * @param template - URL path which may contain slugs: `/foo/:bar`
+ * @param params - Key/value map of parameters to encode into the URL
+ * @returns URL string
+ */
+export function encodeParamsToUrl(template: string, params?: Record<string, any>): string {
+  const keys = Object.keys(params || {}).sort();
+  let out = template
+    .split('/')
+    .map((part) => {
+      if (isSlug(part)) {
+        let key = part.slice(1, -1);
+        if (key.startsWith('...')) {
+          key = key.slice(3);
+          return encodeValue(get(key));
+        } else {
```

Comment:
maybe a bit nit picky but you can remove all `else` keywords here as you have `return`s for both else clauses...